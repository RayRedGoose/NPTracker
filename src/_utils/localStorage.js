export const getItem = item => {
  return JSON.parse(localStorage.getItem(item))
}

export const addItem = (type, item) => {
  return localStorage.setItem(type, JSON.stringify(item))
}

export const addItemToAll = (path, item) => {
  const targetPath = (!getItem(path)) ? [ ] : [...getItem(path)]
  return localStorage.setItem(path, JSON.stringify([...targetPath, item]))
}
