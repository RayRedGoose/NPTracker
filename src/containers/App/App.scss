@import "../../variables.scss";

.app {
  background-color: $bg-color-primary;
  width: 100vw; height: 100vh;
  overflow: hidden;
  main {
    width: 100%; height: 100%;
  }
}

.logo {
  height: 50px;
  opacity: 0.5;
  margin: 20px;
}
