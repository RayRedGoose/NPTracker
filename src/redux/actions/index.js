export const addProcessName = name => ({
  type: 'ADD_PROCESS_NAME',
  name
})

export const addProcessStep = step => ({
  type: 'ADD_PROCESS_STEP',
  step
})

export const increaseStep = () => ({
  type: 'INCREASE_STEP'
})

export const addUserInfo = info => ({
  type: 'ADD_USER_INFO',
  info
})
