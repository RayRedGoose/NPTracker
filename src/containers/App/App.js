import './App.scss'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Redirect } from 'react-router-dom'
import routers from 'routers'

export class App extends Component {
  componentDidMount() {

  }

  render() {
    const { isLogged, process } = this.props
    return (
      <main className="app">
        {(!isLogged || !process)
            ? <Redirect to='/welcome' />
            : <Redirect to='/parks' /> }
        { routers }
      </main>
    );
  }
}

export const mapStateToProps = ({user, isLogged}) => ({
  user, isLogged
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({

  }, dispatch)
)

export default connect(mapStateToProps,mapDispatchToProps)(App);
