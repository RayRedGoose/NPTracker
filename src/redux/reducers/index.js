import { combineReducers } from 'redux'
import { process } from './singleReducers/process'
import { user } from './singleReducers/user'

const rootReducer = combineReducers({
  process,
  user
})

export default rootReducer
