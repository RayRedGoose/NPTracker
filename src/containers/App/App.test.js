import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { addUserInfo } from 'redux/actions'
import { getItem } from '_utils/localStorage'

jest.mock('_utils/localStorage')

describe('App', () => {
  let app, instance, mockProps

  const addUserInfo = jest.fn()
  const user = { name: 'Ray' }
  const process = null

  beforeEach(() => {
    mockProps = {
      user,
      process,
      addUserInfo
    }

    app = shallow(
      <App {...mockProps} />
    );

    instance = app.instance()
  });

  it("should call getItem after component was rendered", () => {
    expect(getItem).toHaveBeenCalledWith('user');
  });

  it("should call addUser method if user info was taken from localStorage after component was rendered", () => {
    getItem.mockImplementation(()=> user)

    const spy = jest.spyOn(instance, 'addUser').mockImplementation()

    instance.componentDidMount()

    expect(spy).toHaveBeenCalledWith(user);
  });

  it("should call addUserInfo prop with user argument if addUser is called", () => {
    instance.addUser(user)

    expect(addUserInfo).toHaveBeenCalledWith(user);
  });

  it("should change isLogged state if addUser is called", () => {
    getItem.mockImplementation(()=> null)

    app.setState({isLogged: false})

    expect(app.state('isLogged')).toEqual(false);

    instance.addUser(user)

    expect(app.state('isLogged')).toEqual(true);
  });
});

describe("mapStateToProps", () => {
  it("should return the object with correct key/value pairs from store", () => {
    const mockStore = {
      user: { },
      process: null,
      parks: []
    }

    const expected = {
      user: { },
      process: null,
    }

    const result = mapStateToProps(mockStore)

    expect(result).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  let mockDispatch, mockProps

  beforeEach(() => {
    mockDispatch = jest.fn()
    mockProps = mapDispatchToProps(mockDispatch);
  });

  it("should call dispatch with an addUserInfo action when addUserInfo is called", () => {
    const user = {name: 'Ray'}

    const actionToDispatch = addUserInfo(user)

    const result = mockProps.addUserInfo(user)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
