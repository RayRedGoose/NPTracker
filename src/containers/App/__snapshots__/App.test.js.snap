// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="app"
>
  <Redirect
    to="/welcome"
  />
  <Switch>
    <Route
      exact={true}
      path="/welcome"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/signup"
      render={[Function]}
    />
  </Switch>
</main>
`;
